assert/partial.go

```diff
@@ -95,48 +95,45 @@ func (assert *Assert) Partial(obj, target interface{}, details ...interface{}) {
 	}
 }
 
-func (assert *Assert) partialStruct(obj reflect.Value, target reflect.Value, details ...interface{}) {
+func (assert *Assert) partialStruct(obj interface{}, target interface{}, details ...interface{}) {
 	assert.t.Helper()
-	objtype := obj.Type()
-	targtype := target.Type()
+	elem := reflect.ValueOf(obj)
+	targ := reflect.ValueOf(target)
```

Comment:
This kind of stuff reminds me of allowing vars to be redeclared with another type (if done safely). I get that it can lead to confusing stuff on some scenarios, but in a lot of cases I end up on this situation where I really don't have another name for the variable (unless I go to adding the type to the name of the var :laughing: ).

Reply:
yeah... annoying